### DISCUSSION TITLE
Help with TV02 (TS0601) TRV valve

### DISCUSSION THREAD

**Discussion by seflm** [Koenkk/zigbee2mqtt#16585] [Devices]
Hello, 

I got the TV02-Zigbee (TS0601) valves from [here](https://www.aliexpress.com/item/1005004861758253.html). Upon installation, I found out that the valves don't do anything unless I change the target temperature by a huge amount.

For example, if the ambient temp is 19 degrees, and I set the target to 22, the valve won't open. I need to set the target to 25-30 for it to open. It just has a huge hysteresis. When closing, the story is the same, I need to set the target 5-8 degrees lower than the current temp for the valve to close. I am thinking there is something wrong, as I don't believe the valve is intended to work in this way. The only time the valves do what it is supposed to, is when I use the boost function. All the pieces I got have the same issue. 

The valves are unusable in this way. Apart from other problems like the auto mode overwriting the off state of the valve. 

It really feels like there is everything wrong with these valves. Do you guys maybe have a similar experience? What are your thoughts?

**Comment by serot23** [comment]
I have exactly the same problem. Now for example couple hours ago i set it on 17,5 and it was then 16 and it is still 16 even on an external sensor. I use the calibration blueprint for this. I have reset and re added in z2m but no difference. Even when it reaches the temperature, the valve stays open and i need to put it in off mode to overrule it.

**Comment by lux73** [comment]
Not a Problem from z2m side - the internal PID algo from Tuya is just crap.

Afaik all you can do is buy another Brand

**Comment by serot23** [comment]
I am trying to do something with an automation or something, so each time when temperature is set manually or something then force boost timer with maximum of the given time which says in the settings but can't figure it out yet and else i think i am going to tru something with an always closed actuator saw a good tutorial somewhere.

**Comment by seflm** [comment]
#16586 is definitely supporting the unusability of these valves. From other threads it seems the internal issues with these valves is partially solved by updating the valve firmware. However OTA is not yet supported by z2m ( #16584 ). Can someone test the valves with the newest firmware? I don't have tuya zigbee hub to retrieve the image.

**Reply by serot23** [reply]
I have connected to the tuya hub, but there are no updates for me.

**Reply by seflm** [reply]
Interesting, according to the #10547. There were some fixes in the firmware. I suppose this is not one of them then.

**Reply by serot23** [reply]
I guess so unfortunately. I am going to try the automation again, this will already solve one part of my problem. Need to make some sort of toggle in the simple thermostat card maybe which triggers the booster

**Comment by serot23** [comment]
I am a dummy jn these things, i have a z2m sonoff controller, maybe if you can explain how i can do it

**Reply by serot23** [reply]
I also have a tuya zigbee hub
